pkg/koordlet/resourceexecutor/executor.go

```diff
@@ -47,29 +53,46 @@ type ResourceUpdateExecutorImpl struct {
 	ResourceCache     *cache.Cache
 	Config            *Config
 
-	onceRun   sync.Once
-	gcStarted bool
+	onceRun       sync.Once
+	gcStarted     bool
+	eventRecorder record.EventRecorder
+	node          *corev1.Node
 }
 
-var singleton = &ResourceUpdateExecutorImpl{
-	ResourceCache: cache.NewCacheDefault(),
-	Config:        Conf,
+var singleton *ResourceUpdateExecutorImpl
+var recorderSingleton record.EventRecorder
+var nodeSingleton *corev1.Node
+
+func NewResourceExecutorEventRecorder(schema *apiruntime.Scheme, kubeClient kubernetes.Interface, nodeName string, node *corev1.Node) {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(&clientcorev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
+	recorder := eventBroadcaster.NewRecorder(schema, corev1.EventSource{Component: "koordlet-resource-executor", Host: nodeName})
+	recorderSingleton = recorder
+	nodeSingleton = node
 }
 
 func NewResourceUpdateExecutor() ResourceUpdateExecutor {
+	if singleton == nil {
```

Comment:
need to resolve the concurrency problem
Why not just initialize the event broadcaster in the definition of the executor?

e.g.

```go
var singleton = &ResourceUpdateExecutorImpl{
  eventBroadCaster: record.NewBroadcaster(),
  // ...
}
```